app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -687,21 +687,26 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
      * @param searchEngines List of [SearchEngine] to display.
      */
     private fun updateSearchSelectorMenu(searchEngines: List<SearchEngine>) {
-        searchSelectorMenu.menuController.submitList(
-            searchSelectorMenu.menuItems() +
-                searchEngines
-                    .reversed()
-                    .map {
-                        TextMenuCandidate(
-                            text = it.name,
-                            start = DrawableMenuIcon(
-                                drawable = it.icon.toDrawable(resources)
-                            )
-                        ) {
-                            interactor.onMenuItemTapped(SearchSelectorMenu.Item.SearchEngine(it))
-                        }
-                    }
-        )
+        val searchEngineList = searchEngines
+            .map {
+                TextMenuCandidate(
+                    text = it.name,
+                    start = DrawableMenuIcon(
+                        drawable = it.icon.toDrawable(resources)
+                    )
+                ) {
+                    interactor.onMenuItemTapped(SearchSelectorMenu.Item.SearchEngine(it))
+                }
+            } + searchSelectorMenu.menuItems()
+
+        when (requireContext().settings().shouldUseBottomToolbar) {
+            true -> {
+                searchSelectorMenu.menuController.submitList(searchEngineList.reversed())
+            }
+            false -> {
+                searchSelectorMenu.menuController.submitList(searchEngineList)
+            }
+        }
```

Comment:
```suggestion
       searchSelectorMenu.menuController.submitList(
           if (requireContext().settings().shouldUseBottomToolbar) searchEngineList.reversed() else searchEngineList
       )
```

Reply:
Sure, but I don't think this will pass ktlint if there is no brackets.  I'll simplify this (maybe remove the brackets with "when")

Reply:
As expected, detekt did not like if else statement with no brackets.  I've removed the brackets in the "when" statement so it's more readable.  Thanks

Reply:
I moreso was suggesting to replace the `when` with an `if`, since it's just checking a boolean. You could still do:

```
      searchSelectorMenu.menuController.submitList(
           if (requireContext().settings().shouldUseBottomToolbar) {
               searchEngineList.reversed() 
           } else {
              searchEngineList
           }
       )
```

Reply:
Not going to block a PR for a personal code style nit, so let's keep this moving